packages/docusaurus-plugin-google-gtag/src/index.js

```diff
@@ -7,12 +7,72 @@
 
 const path = require('path');
 
-module.exports = function() {
+module.exports = function(context) {
+  const {siteConfig} = context;
+  const {themeConfig} = siteConfig;
+  const {gtag} = themeConfig || {};
+
+  if (!gtag) {
+    throw new Error(
+      `You need to specify 'gtag' object in 'themeConfig' with 'trackingId' field in it to use docusaurus-plugin-google-gtag`,
+    );
+  }
+
+  const {trackingID} = gtag;
+
+  if (!trackingID) {
+    throw new Error(
+      'You specified the `gtag` object in `themeConfig` but the `trackingID` field was missing. ' +
+        'Please ensure this is not a mistake.',
+    );
+  }
+
+  const isProd = process.env.NODE_ENV === 'production';
   return {
     name: 'docusaurus-plugin-google-gtag',
 
     getClientModules() {
-      return [path.resolve(__dirname, './gtag')];
+      return isProd ? [path.resolve(__dirname, './gtag')] : [];
+    },
+
+    injectHtmlTags() {
+      if (!isProd) {
+        return {};
+      }
+      return {
+        headTags: [
+          {
+            tagName: 'link',
```

Comment:
Why do we still need preconnect to https://www.google-analytics.com?

I see `<script type="text/javascript" async="" src="https://www.google-analytics.com/analytics.js"></script>` in the deploy preview's HTML which seems to be added in by the loaded gtag script. Is that why you do preconnect to google-analytics.com here? If so, it would be good to add a comment as it isn't obvious.

Reply:
yes. because gtag automatically add google analytics script for you. So its better to preconnect